### PR TITLE
fix Readme grammar

### PR DESCRIPTION
**Pull request by herou**
Some grammar errors are fixed.

### CODE REVIEW

**Comment by herou** [comment]
@gmale I fixed some grammar errors that I found in the documentation.

File: README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ This lightweight SDK connects Android to Zcash. It welds together Rust and Kotli
 
 ## Requirements
 
-This SDK is designed to work using a light client server [lightwalletd](https://github.com/zcash-hackworks/lightwalletd)
+This SDK is designed to work using a light client-server [lightwalletd](https://github.com/zcash-hackworks/lightwalletd)
```
Comment (gmale) [inline README.md:35]: This is a server for "light clients" so a hyphen here wouldn't be needed but maybe we should reword it so it's more clear? It is probably fine to shorten this to:
 
>This SDK is designed to work with [lightwalletd](https://github.com/zcash/lightwalletd)

File: README.md [hunk 2]
```diff
@@ -61,7 +61,7 @@ The [Synchronizer](docs/-synchronizer/README.md) takes care of
     - Downloading the latest compact blocks in a privacy-sensitive way
     - Scanning and trial decrypting those blocks for shielded transactions related to the wallet
     - Processing those related transactions into useful data for the UI
-    - Sending payments to a full node through the light client server
+    - Sending payments to a full node through the light client-server
```
Comment (gmale) [inline README.md:64]: Since this is a server for "light clients" we could probably just change this to:
> Sending payments to a full node through [lightwalletd](https://github.com/zcash/lightwalletd)

**Review by gmale — CHANGES_REQUESTED** [review]
Thanks! This is very much appreciated. One of the edits is probably moreso a sign that the language isn't clear. I've added details in the review comments.

**Comment by herou** [comment]
@gmale Committed your suggestions.
